koku/masu/processor/ocp/ocp_report_db_cleaner.py

```diff
@@ -147,3 +148,27 @@ def purge_expired_report_data_by_date(self, expired_date, simulate=False):
                 LOG.info(log_json(msg="deleted table partitions", count=del_count, schema=self._schema))
 
         return removed_items
+
+    def purge_expired_trino_partitions(self, expired_date, provider_uuid=None, simulate=False):
+        """Removes expired trino partitions."""
+        LOG.debug(f"purge_expired_trino_partitions: {expired_date}, {provider_uuid}, {simulate}")
+        if expired_date is None and provider_uuid is None:
+            err = "This method must be called with expired_date or provider_uuid"
+            raise OCPReportDBCleanerError(err)
+
+        with OCPReportDBAccessor(self._schema) as accessor:
+            for table, source_column in EXPIRE_MANAGED_TABLES.items():
+                LOG.debug(f"{table}, {source_column}")
+                results = accessor.find_expired_trino_partitions(table, source_column, str(expired_date.date()))
+                if results:
+                    LOG.info(f"Discovered {len(results)} expired partitions")
+                else:
+                    LOG.info("No expired partitions")
+                    return
+                if simulate:
+                    for partition in results:
+                        LOG.info(f"partition_info: {partition}")
+                else:
+                    for result in results:
+                        year, month, source_value = result
+                        accessor.delete_hive_partition_by_month(table, source_value, year, month, source_column)
```

Comment:
Total nitpick, but this if else isn't totally necessary:
```suggestion
                for partition in results:
                    LOG.info(f"partition_info: {partition}")
                    if not simulate:
                        year, month, source_value = partition
                        accessor.delete_hive_partition_by_month(table, source_value, year, month, source_column)
```